pkg/algorithms/edivisive/edivisive.py

```diff
@@ -31,9 +33,47 @@ def _analyze(self):
         # filter by direction and ack'ed issues
         for metric, changepoint_list in change_points_by_metric.items():
             for i in range(len(changepoint_list)-1, -1, -1):
-                if ((self.metrics_config[metric]["direction"] == 1 and changepoint_list[i].stats.mean_1 > changepoint_list[i].stats.mean_2) or (self.metrics_config[metric]["direction"] == -1 and changepoint_list[i].stats.mean_1 < changepoint_list[i].stats.mean_2) or (str(changepoint_list[i].index) + "_" + changepoint_list[i].metric in ackSet)):
+                deleted = False
+                if (self._has_changepoint(metric, changepoint_list, i) or
+                    self._is_acked(ackSet, changepoint_list, i) or
+                    self._is_under_threshold(metric, changepoint_list, i)):
+                    deleted=True
                     del changepoint_list[i]
+                if (not deleted and self.metrics_config[metric]["depends_on"] != ""):
+                    has_depending_changepoint = self._depending_metric_has_chagepoint(change_points_by_metric,
+                                                                                      ackSet,
+                                                                                      metric,
+                                                                                      changepoint_list[i].index)
+                    if not has_depending_changepoint:
+                        del changepoint_list[i]
 
         if [val for li in change_points_by_metric.values() for val in li]:
             self.regression_flag=True
         return series, change_points_by_metric
+
+
+
+    def _depending_metric_has_chagepoint(self, change_points_by_metric: Dict[str, List[ChangePoint]], ackSet, metric, index) -> bool:
+        depending_metric = self.metrics_config[metric]["depends_on"]
+        changepoint_list = change_points_by_metric[depending_metric]
+        for i in range(len(changepoint_list)-1, -1, -1):
+            if changepoint_list[i].index == index:
```

Comment:
By any chance is this a check to see if the metric and its dependant have changepoint listed on the same index implying a changepoint on same date?

Reply:
Yes, it is.